calico fixes

disconn3ct opened the pull request:
<!--
Before submitting a pull request:
- Please ensure the target branch is "dev" (active development): https://github.com/MichaIng/DietPi/tree/dev
- Please ensure changes have been tested and verified functional.
-->
Add more missing configs for Calico. With this config it joins the mesh successfully and seems to be working normally.

disconn3ct committed:
chore: diffconfig

MichaIng commented on .build/images/Quartz64/quartz64_defconfig:
```diff
@@ -44,6 +44,7 @@ CONFIG_BLK_DEV_NBD=m
 CONFIG_BLK_DEV_NVME=y
 CONFIG_BLK_DEV_RAM=n
 CONFIG_BLK_DEV_SD=y
+CONFIG_BLK_DEV_THROTTLING=y
```
That one seems to be pretty special, are you sure it is really used? E.g. the RPi kernel does not define it either. Probably it requires hardware support as well.

MichaIng commented on .build/images/Quartz64/quartz64_defconfig:
```diff
@@ -336,6 +341,12 @@ CONFIG_IP_SET_HASH_NETPORTNET=m
+CONFIG_IP_VS_RR=m
+CONFIG_IP_VS_NFCT=y
+CONFIG_IP_VS_FTP=n
+CONFIG_IP_VS_PE_SIP=n
```
Why do you disable those explicitly? They are not defined (not even as "is not set") in our final kernel config, so it would not change something, but RPi and Debian kernels have the modules for those enabled. So, I guess either the lines can be removed, or we add the modules as well.

MichaIng commented on .build/images/Quartz64/quartz64_defconfig:
```diff
@@ -661,7 +673,7 @@ CONFIG_NFT_SOCKET=m
 CONFIG_NFT_TPROXY=m
 CONFIG_NFT_TUNNEL=m
-CONFIG_NFT_XFRM=m
+CONFIG_NFT_XFRM=n
```
Same here, this is set in RPi and Debian kernel. So disabling it explicitly, might beak something. If this module must not be loaded in your case, blacklist it via `/etc/modprobe.d`.

MichaIng commented on .build/images/Quartz64/quartz64_defconfig:
```diff
@@ -1122,3 +1134,16 @@ CONFIG_ZSWAP=y
 CONFIG_ZSWAP_COMPRESSOR_DEFAULT_ZSTD=y
 CONFIG_ZSWAP_DEFAULT_ON=y
 CONFIG_ZSWAP_ZPOOL_DEFAULT_ZSMALLOC=y
+CONFIG_RT_GROUP_SCHED=y
```
Same as `CONFIG_BLK_DEV_THROTTLING`, this group scheduler is not available on RPi or Debian kernel either, so seems to be a very rare use case. Are you sure if is needed for Calico networks?

MichaIng commented on .build/images/Quartz64/quartz64_defconfig:
```diff
@@ -1122,3 +1134,16 @@ CONFIG_ZSWAP=y
+CONFIG_RT_GROUP_SCHED=y
+CONFIG_INET_ESP=m
+CONFIG_INET_ESPINTCP=n
+CONFIG_INET_ESP_OFFLOAD=n
```
Both are not defined in our kernel, so it shouldn't be required to explicitly disable them.

MichaIng commented on .build/images/Quartz64/quartz64_defconfig:
```diff
@@ -1122,3 +1134,16 @@ CONFIG_ZSWAP=y
+CONFIG_XFRM_INTERFACE=n
+CONFIG_XFRM_MIGRATE=n
+CONFIG_XFRM_STATISTICS=n
+CONFIG_XFRM_SUB_POLICY=n
```
Same here, why is it needed to explicitly disable those features? Or is it just that they get implicitly enabled with `CONFIG_XFRM=y`, and your aim was to enable only the minimal additional feature set?

MichaIng committed:
v9.9

- Quartz64 | Remove features which are not available on Debian and RPi kernel either, and do not disable features explicitly, which are not disabled in Debian or RPi kernel explicitly as well. For sub features, we better follow Linux defaults, and how it is on other distros and images, to not cause confusion with a disabled sub feature, which is commonly enabled and expected. And we do not want to have an overly extended kernel. If a feature is not even enabled in the rich default Debian and/or RPi kernel, then it should be better understood, what for it is really needed.
